mongoexport/main/mongoexport.go

```diff
@@ -62,83 +36,12 @@ func main() {
 		return
 	}
 
-	// verify uri options and log them
-	opts.URI.LogUnsupportedOptions()
-
-	provider, err := db.NewSessionProvider(*opts)
-	if err != nil {
-		log.Logvf(log.Always, "%v", err)
-		os.Exit(util.ExitError)
-	}
-	defer provider.Close()
-
-	// temporarily allow secondary reads for the isMongos check
-	provider.SetReadPreference(mgo.Nearest)
-	isMongos, err := provider.IsMongos()
+	exporter, err := mongoexport.New(opts)
 	if err != nil {
 		log.Logvf(log.Always, "%v", err)
 		os.Exit(util.ExitError)
 	}
-
-	provider.SetFlags(db.DisableSocketTimeout)
```

Comment:
Based on the new driver code and talking to Kris, I think connections will have no explicit socket timeout if we don't explicitly set one when create the client, so this line isn't needed.

Reply:
There's likely no value to keeping this in. There will need to be some follow-up work to remove unused functions in common after all of the ports are done, but I think that should be done as a separate commit/ticket rather than part of this port.

Reply:
sounds good to me. is there a ticket for that?

Reply:
https://jira.mongodb.org/browse/TOOLS-2248